solutions/bronze/usaco-1276.mdx

```diff
@@ -0,0 +1,211 @@
+---
+id: usaco-1276
+source: USACO Bronze 2023 January
+title: Air Cownditioning II
+author: Chuyang Wang
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_prob2_bronze_jan23.html)
+
+## Explanation
+
+Since $M = 10$ is given, we can iterate through all $2^{10}$ possible subsets of
+air conditioners. For each of them, we check if this subset of air conditioners
+satisfy the requirements of the cows by iterating through all possible positions
+$i \in [1;100]$. For each position $i$, we can iterate through all available air
+conditioners in the current subset to find out the temperature reduced at $i$
+and iterate through all cows to find the cow at this position and its demand. If
+the requirement at each position is fulfilled, we update the minimal cost
+accordingly.
+
+There are two ways to generate all subsets of air conditioners: Using recursion
+or a bitmask.
+
+
+## Solution 1: Subset with Recursion
+
+**Time Complexity:** $\mathcal{O}(2^M \cdot 100 \cdot (M + N))$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int N, M;
+// {s, t, c}
+vector<array<int, 3>> cows;
+// {a, b, p, m}
+vector<array<int, 4>> air_conditioners;
```

Comment:
use structs

Reply:
I seriously do not think struct provides any clearity here: we are using auto structured binding already. Declaring two extra structs just makes the code unnecessarily longer

Reply:
if you insits